### ISSUE TITLE
TextTheme Color not Working in V7

### ISSUE THREAD

**Issue by theTello** [rydmike/flex_color_scheme#151]
After upgrading to v7 of flex color scheme, text colors defines for various TextThemes don't get applied and I have 

<img width="575" alt="image" src="https://github.com/rydmike/flex_color_scheme/assets/75843984/153e5776-ef41-4f0b-b26f-82fc5d457478">

In order to get it to work I have to specify the color as seen below. This wasn't the issue in V6 of flex color scheme however.

<img width="575" alt="image" src="https://github.com/rydmike/flex_color_scheme/assets/75843984/bd4dce84-b90e-4c0e-8428-ffd82bff4185">

**Comment by rydmike** [comment]
Hi @theTello,

Thanks for the issue report.

Can you provide the above `TextTheme` snippet as code in a dart code block so I can copy it and investigate further?

```dart
textTheme: TextTheme(
  titleLarge: TextStyle(...),
)
```

**Comment by theTello** [comment]
Hello @rydmike 
Sure, Below is the snippet

```

 textTheme: TextTheme(

  /// Default style for buttons in light theme
  titleLarge: GoogleFonts.dmSans(
    fontSize: 28,
    fontWeight: FontWeight.w700,
    height: 32 / 28,
    color: const Color(0xFF2D3436),
  ),

    /// Default style for buttons in light theme
    titleMedium: GoogleFonts.dmSans(
      fontSize: 24,
      fontWeight: FontWeight.w700,
      height: 28 / 24,
      color: const Color(0xFF2D3436),
    ),

    /// Default style for buttons in light theme
    titleSmall: GoogleFonts.dmSans(
      fontSize: 20,
      fontWeight: FontWeight.w500,
      height: 24 / 20,
      color: const Color(0xFF2D3436),
    ),

    /// Default style for buttons in light theme
    labelLarge: GoogleFonts.dmSans(
      fontSize: 18,
      fontWeight: FontWeight.w500,
      height: 20 / 18,
      color: Colors.white,
    ),

    /// Default style for Large body texts
    bodyLarge: GoogleFonts.dmSans(
      fontSize: 16,
      fontWeight: FontWeight.w400,
      height: 20 / 16,
      color: const Color(0xFF2D3436),
    ),

    /// Default style for small body texts
    bodyMedium: GoogleFonts.dmSans(
      fontSize: 14,
      fontWeight: FontWeight.w400,
      height: 18 / 14,
      color: const Color(0xFF636E72),
    ),

    /// Default style for small body texts
    bodySmall: GoogleFonts.dmSans(
      fontSize: 12,
      fontWeight: FontWeight.w400,
      height: 16 / 12,
      color: const Color(0xFF636E72),
    ),
  ),


```

**Comment by rydmike** [comment]
Thanks, I will look into tonight when I'm not at the day job 😄

**Comment by rydmike** [comment]
Confirmed, it is a regression bug. I thought I had a test case for this usage, but apparently not. Will be adding tests that will hit and test this use case and then applying a fix.

Are you OK with using latest Flutter 3.10 release? Because I probably wont't be back porting this fix to FlexColorScheme v7.0.x. It will only come to latest as FlexColorScheme 7.1.1.

Nice find and thanks for the issue report.

**Comment by theTello** [comment]
Okay, I'm comfortable with flutter 3.10.0, Really appreciate all the efforts you're putting in, thank you so much

**Comment by rydmike** [comment]
Sorry the fix is taking a bit longer than I thought. It turned out to be a bit bigger than I anticipated. 

Would have been easy to just revert. However, I also wanted to keep the feature that automatically give GoogleFont's default textTheme the correct contrast color automatically in FlexColorScheme, while also of course allowing for custom colors, like your case.

I have it all figured out and fixed now, with test for the regression and the new revised GoogleFonts default TextTheme color fix. The test for the regression and additional workaround for the `GoogleFonts` default color convenience in FCS were also quite extensive. But glad to have those tests added now.

While doing all this, I also started thinking that `GoogleFonts` package should maybe do a better job, and not force the color from `ThemeData.light().textTheme` as its default color in a `TextTheme`, it should hand that over to ThemeData to apply.

Well I say perhaps, because if/when it is used in ThemeData created, it needs a color. Still going to raise an issue that maybe there should be a new null option suitable for use in `ThemeData` and its `textTheme` and `primaryTextTheme` because the current one is not. Due to the way the its default is based on `ThemeData.light().textTheme` its styles do not have fontSize now, so that might also support that it should not have color either. Size comes via the localization step, it is not applied yet in `ThemeData` alone.

Anyway, I want to have the `GoogleFonts` raised issue referenced in the change-log for this fix as well. So next step is to raise that issue. Then merge this fix, and publish new package version 7.1.1 of FCS.

If `GoogleFonts` will change their default color is another matter, if they do then `GoogleFonts` TextTheme will work better also when used with vanilla `ThemeData` and its `textTheme` and `primaryTextTheme` properties. If that happen I can later remove some of my own version of the thing that detect if it if a GoogleFonts default `TextTheme` and null's its colors, so it get correct color and contrast defaults.

**Comment by rydmike** [comment]
I have now also added the described GoogleFonts color issue to the GoogleFonts repo. The one about the strange hard coded color and need to work around it when using it in ThemeData to get correct default colors. I added a ref to it in the change log for this issue as well.

The GoogleFonts issue can be found here https://github.com/material-foundation/flutter-packages/issues/401 if you want to take a look.

I have not idea if it is something Material team is willing to change. It is not relevant for the solution of this issue, only for my workaround to deal with the GoogleFonts color issue. I do however think my workaround can remain as is as well. Plus it is another convenience of FlexColorScheme when combined with GoogleFonts, that vanilla ThemeData users will not have, unless Material team fixes/change GoogleFonts as I suggested in the issue.


I am ready to release this fix. will do this evening, after some sleep, so I can double check everything once more with fresh eyes. But it is ready to go 😄
